RUFAS/routines/field/manager/schedule.py

```diff
@@ -141,17 +99,173 @@ def _validate_years(years: List[int]) -> bool:
         """
         return all(0 < years[index] <= years[index + 1] for index in range(0, len(years) - 1))
 
+    def generate_events(
+        self,
+        years: list[int],
+        days: list[int],
+        additional_attributes: Optional[list[Any]],
+        additional_attributes_events: list[list[Any]],
+        event_class: Any,
+        pattern_skip: int,
+        pattern_repeat: int,
+        heat_scheduled_harvest: bool,
+    ) -> list[Any]:
+        """
+        Generic method to generate application events.
+
+        Parameters
+        ----------
+        years : List[int]
+            List of years for the schedule.
+        days : List[int]
+            List of days for the schedule.
+        additional_attributes : List[List]
+            Additional general attributes for the events (e.g., crop reference).
+        additional_attributes_events : List[List]
+            Additional attributes for each of the events (e.g., nitrogen_mass, phosphorus_mass, etc.).
+        event_class : class
+            The class to instantiate for each event.
+        pattern_skip : int
+            Number of years to skip.
+        pattern_repeat : int
+            Number of times the pattern should repeat.
+        heat_scheduled_harvest : bool
+            Flag indicating if heat unit scheduling is utilized for harvesting decisions.
+
+        Returns
+        -------
+        list
+            List of instantiated event objects.
+
+        """
+        all_years = self.repeat_pattern(years, pattern_skip, pattern_repeat)
+        all_days = days * (pattern_repeat + 1)
+        repeated_attributes = [attr * (pattern_repeat + 1) for attr in additional_attributes_events]
+        all_events = list(zip(all_years, all_days, *repeated_attributes))
+        if heat_scheduled_harvest:
+            all_events[:] = [harvest for harvest in all_events if harvest[2] in FINAL_HARVEST_OPERATIONS]
+        result = [event_class(*additional_attributes, *event) for event in all_events]
+
+        return result
+
     @staticmethod
-    def _repeat_pattern(pattern: List[int], skip: int = 0, repeat: int = 0) -> List[int]:
+    def validate_depths(depths: list[float]) -> bool:
```

Comment:
```suggestion
    def validate_positive_values(values: list[float]) -> bool:
```
By changing this method to not be specific to depths, it makes clearer that it could be applicable to any values that need to be positive.